lib/ui/text.dart

```diff
@@ -1501,6 +1545,40 @@ class ParagraphBuilder extends NativeFieldWrapperClass2 {
   }
   String _addText(String text) native 'ParagraphBuilder_addText';
 
+  /// Adds an inline placeholder space to the paragraph.
+  ///
+  /// The paragraph will contain a rectangular space with no text of the dimensions
+  /// specified.
+  ///
+  /// The [width] and [height] parameters specify the size of the rectangle, and
+  /// [baseline] indicates the distance of the alphabetic baseline from the top of
+  /// of the rectangle. This determines how the rectangle is aligned to the
+  /// surrounding text.
+  ///
+  /// For example, to make a 50x50 empty rectangle 'hole' that sits completely on
+  /// top of the alphabetic baseline, pass in `addPlaceholder(50, 50, 50)`. For
+  /// a box that is vertically centered on the alphabetic baseline, pass in
+  /// `addPlaceholder(50, 50, 25);`.
```

Comment:
we should make sure not to default to alphabetic in the API (even if our implementation doesn't yet do this right)